yb-voyager/cmd/live_migration.go

```diff
@@ -65,14 +70,14 @@ func streamChanges() error {
 		}
 		log.Infof("got next segment to stream: %v", segment)
 
-		err = streamChangesFromSegment(segment, evChans, processingDoneChans)
+		err = streamChangesFromSegment(segment, evChans, processingDoneChans, eventChannelsMetaInfo)
 		if err != nil {
 			return fmt.Errorf("error streaming changes for segment %s: %v", segment.FilePath, err)
 		}
 	}
 }
 
-func streamChangesFromSegment(segment *EventQueueSegment, evChans []chan *tgtdb.Event, processingDoneChans []chan bool) error {
+func streamChangesFromSegment(segment *EventQueueSegment, evChans []chan *tgtdb.Event, processingDoneChans []chan bool, eventChannelsMetaInfo map[int]tgtdb.EventChannelMetaInfo) error {
```

Comment:
Why use `map` instead of a `list`?

Reply:
Use-case is to look up the metainfo(vsn) on channel, so map seemed a better fit than a list. Alternative is: I could also just return a list (ordered by vsn from db itself), and lookup by index, but this seemed like a more deterministic way.